packages/workspace/src/tasks-runner/task-orchestrator.ts

```diff
@@ -141,50 +194,161 @@ export class TaskOrchestrator {
   }
 
   private async runTasks() {
```

Comment:
This needs to be broken down. It's to follow.

In general, what do you think about splitting TaskOrchestrator into several parts. 

E.g., the part that forks the processes, the part that manipulates the graph, etc..

Right now it's all together and it's hard to follow.